src/Halogen/Aff/Driver.purs

```diff
@@ -154,8 +154,23 @@ runUI renderSpec component = do
           handleLifecycle \lchs -> render lchs ref
           pure a
     Subscribe es next -> do
-      let consumer = forever (lift <<< evalF ref =<< CR.await)
-      forkAff $ CR.runProcess (unwrap es $$ consumer)
+      forkAff do
+        { producer, done } <- ES.unEventSource es
+        isDone <- liftEff $ newRef false
+        blocker <- AV.makeVar
+        let
+          consumer :: CR.Consumer (f' Boolean) (Aff (HalogenEffects eff)) Unit
+          consumer = flip tailRecM unit \_ -> do
```

Comment:
`tailRecM` is unnecessary in Co/FreeT since it already requires a `MonadRec` constraint to interpret it. This is just adding a multiplier of binds.